modules/ecash-lib/src/unsignedTx.ts

```diff
@@ -83,6 +90,102 @@ export interface SighashPreimage {
     redeemScript: Script;
 }
 
+// Write the legacy preimage used pre-UAHF.
+// It's modelled closely after SignatureHash in interpreter.cpp.
+function writeLegacyPreimage(
+    writer: Writer,
+    tx: Tx,
+    scriptCode: Script,
+    inputIdx: number,
+    sigHashType: SigHashType,
+) {
+    const hasAnyoneCanPay =
+        sigHashType.inputType == SigHashTypeInputs.ANYONECANPAY;
+
+    const writeLegacyScriptCode = () => {
```

Comment:
any advantage of keeping these function definitions outside of `writeLegacyPreImage`?

mb not used anywhere else but I'm not familiar with the rationale for keeping them all in this function, vs defining them outside then using them here.

Reply:
The reason is that these are highly specific functions that really should never be used outside.

It also mimics the C++ code a little where these are encapsulated all under one class.